test/fixture/bad.ts

```diff
@@ -1,3 +0,0 @@
-function bad(): string {
```

Comment:
Because there is no way to tell this lib something should fail by design and this is, by design, bad TypeScript code. Including this causes an error in the tests:

```sh
➜  jest-runner-tsc git:(master) ✗ yarn test
yarn run v1.13.0
$ npm run-script build && jest

> jest-runner-tsc@0.0.0-development build /Users/jeroenc/dev/jest-runner-tsc
> babel src -d dist

Successfully compiled 2 files with Babel.
 PASS   lint  src/index.js
 PASS   lint  ./jest.config.js
 PASS   lint  test/index.test.js
 PASS   lint  src/runTsc.js
 FAIL   tsc  test/lib.test.ts
Type '42' is not assignable to type 'string'.
   7 | describe('compiling incorrect TypeScript should fail', () => {
   8 |   const libTestFail = (): string => {
>  9 |     return 42;
     |     ^^^^^^^^^^
  10 |   }
  11 | });
 PASS   test  test/index.test.js

Test Suites: 1 failed, 5 passed, 6 total
Tests:       1 failed, 5 passed, NaN total
Snapshots:   0 total
Time:        6.436s
Ran all test suites in 3 projects.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

This is after modifying `lib.test.ts` to:

```ts
describe('compiling correct TypeScript should work', () => {
  const libTest = (): string => {
    return 'apples';
  };
});

describe('compiling incorrect TypeScript should fail', () => {
  const libTestFail = (): string => {
    return 42;
  }
});
```

So unless the lib gets some feature where we can tell Jest that a test should fail it is not possible to test the lib with its own runner while also having a test that should intentionally fail.

This would also happen before when you'd run `jest -c jest.tsc.config.js`, however now that I have made it so all runners are always ran it happens always.

Note that this therefore assumes that the lib itself also runs over its own test code which I personally thing it really should do. Just look at projects like TypeScript that's written entirely *in* TypeScript or Webpack which packs itself with webpack.

Reply:
Can we simply add them to ignore paths in jest config?

Reply:
Sure we can but then that would be the same end goal as excluding entirely. I mean ignored tests are basically useless. Sure it might be a nice example for other people, but I'm quite sure that anyone starting TypeScript wouldn't make such a mistake let alone need an example in this lib to have them understand it's wrong 

(Considering both running tsc or ts-node will crash on this type of bad code)